polkadot/cli/src/lib.rs

```diff
@@ -53,10 +61,12 @@ use std::sync::mpsc;
 /// 9556-9591		Unassigned
 /// 9803-9874		Unassigned
 /// 9926-9949		Unassigned
-pub fn run<I, T>(args: I, exit: mpsc::Receiver<()>) -> error::Result<()> where
+pub fn run<I, T>(args: I) -> error::Result<()> where
 	I: IntoIterator<Item = T>,
 	T: Into<std::ffi::OsString> + Clone,
 {
+	let mut core = reactor::Core::new().expect("tokio::Core could not be created");
```

Comment:
can't we propagate this and other errors upwards instead of panicking?

Reply:
is there any reasonable scenario that this could happen when running on an supported platform? it looks rather like it's not an error you'd ever have to worry about (or even is possible to) handle. if so, i think a simple panic is better since it doesn't pollute the rest of the codebase with basically useless error infrastructure.

Reply:
no idea under exactly what circumstances this can fail. Too many open file descriptors is one possibility.

Reply:
There's no way to reasonably handle this error, other than terminate anyway